lib/api/1.1/northbound/login.js

```diff
@@ -0,0 +1,75 @@
+// Copyright 2016, EMC, Inc.
+
+'use strict';
+
+var di = require('di'),
+    express = require('express'),
+    passport = require('passport');
+
+module.exports = loginRouterFactory;
+
+di.annotate(loginRouterFactory, new di.Provide('Http.Api.Login'));
+di.annotate(loginRouterFactory, new di.Inject(
+        'Auth.Services'
+    )
+);
+
+function loginRouterFactory (
+    authService
+) {
+    var BAD_REQUEST_STATUS = 400;
+    var UNAUTHORIZED_STATUS = 401;
+    var ERROR_STATUS = 500;
+
+    var router = express.Router();
+
+    /**
+     * @api {post} /api/1.1/login/ POST /
+     * @apiVersion 1.1.0
+     * @apiDescription get a token before accessing other apis
+     * @apiName login-post
+     * @apiGroup login
+     * @apiParam {String} username username used to login
+     * @apiParam {String} password password used to login
+     * @apiSuccess {json} a token to be used when accessing other apis:
+     *      {
+     *       "token": "eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJ1c2VyIjoiY
+     *       WRtaW4iLCJpYXQiOjE0NTI0ODA3MTIsImV4cCI6MTQ1MjU2NzExMn0.Qo-9y
+     *       omlkd0WE1jlEA-L1WqA6k_9qJYhkicQL5IUmSg"
+     *      }
+     */
+
+    router.post('/login',
+        function(req, res, next) {          //Middleware to do authentication.
+            passport.authenticate('local', {
+                    session: false
+                },
+                function(err, user, challenges){ // Callback to handle auth error
+                    if(err){
+                        res.status(ERROR_STATUS).send({message: 'Internal server error'});
+                    }
+                    else if (!user){
+                        if(challenges.message === 'Missing credentials'){
+                            res.status(BAD_REQUEST_STATUS).send(challenges);
```

Comment:
@yyscamper passport.authenticate returns a connect/express middleware function.  In this case err indicates a problem with the authentication process, not failed authentication so generically mapping this to 500 status code seems reasonable to me.  I think these two conditions can be combined into one:

                       else if(challenges.message === 'Invalid username or password') {
                             res.status(UNAUTHORIZED_STATUS).send(challenges);
                        }
                        else {
                            res.status(UNAUTHORIZED_STATUS).send(challenges);
                       }